subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -49,27 +49,21 @@ This is a potentially breaking change if you are consuming the console output of
 
 === Deprecations
 
-[[custom_configuration_roles]]
-==== Custom configuration roles
-
-// TODO: There is currently no API for users to provide "custom roles" or even
-// any roles at all other than the existing role-mutating methods. We should revisit
-// this deprecation log and section when we introduce a public API for creating
-// locked-role configurations.
-
-Custom roles have been deprecated.
-Use a pre-defined role instead.
-
 [[configurations_allowed_usage]]
 ==== Allowed configurations usage
 
-The usage of configurations should be fixed at creation.
-Mutating the allowed usage on a configuration is deprecated.
+The allowed usage of a configuration should be immutable after creation.
+Mutating the allowed usage on a configuration created by a Gradle core plugin is deprecated.
 This includes calling any of the following `Configuration` methods:
 
 - `setCanBeConsumed(boolean)`
 - `setCanBeResolved(boolean)`
 
+This will now emit a deprecation warning, except for certain special cases which deal with legacy configurations or make allowances for the existing behavior of popular plugins.
+Custom configurations created by non-core plugins should not be affected by this change.
```

Comment:
```suggestion
Configurations created by non-core plugins should not be affected by this change.
```

I would avoid the word custom here, since I wouldn't consider configurations created by third-party plugins to be "custom"